Document alerts and notifications for NOM 1.6 (#106)
neo4j/docs-ops-manager: modules/ROOT/pages/monitoring/concepts.adoc

= Monitoring concepts
:description: This section describes the concepts that are important to understand how Neo4j Ops Manager can be used for monitoring systems.

> This section describes the concepts that are important to understand how Neo4j Ops Manager can be used for monitoring systems in an optimal way.

Reviewer: ```suggestion
This section describes important concepts for understanding how to optimally use Neo4j Ops Manager to monitor systems.
```

== Conditions

> A **Condition** specifies the constraints when the monitored entity reaches a certain state.

Reviewer: Again with the caps. Unless you are changing the commonly accepted usage of the word _condition_, I'd recommend to not capitalize.

> Conditions are periodically evaluated, evaluation results are persisted.

Reviewer: ```suggestion
Conditions are periodically evaluated, and evaluation results are persisted.
```

There are two **types** of Conditions:

> * **Event-based Conditions**: a Condition evaluates to **true** when a specific event was generated by the monitoring subsystem.

Reviewer: ```suggestion
* **Event-based Conditions**: a Condition evaluates to **true** when a specific event is generated by the monitoring subsystem.
```

Examples: "Unreachable/Offline agents detected", "Incompatible agent connected".

> * **Metric-based Conditions**: a Condition evaluates to **true** when the metric(s) assigned to it change in a way, so that its expression evaluates to **true**.

Reviewer: ```suggestion
* **Metric-based Conditions**: a Condition evaluates to **true** when the metric(s) assigned to it change in a way so that its expression evaluates to **true**.
```

Example: "Host CPU usage exceeds 80% longer than 2 minutes".

> A Condition can have a **grace time**. If present, grace time means that this Condition would only evaluate to **true** if all its **Evaluations** during the grace time evaluated to **true**.

Reviewer: ```suggestion
A Condition can have a **grace time**. 
If present, grace time means that this Condition only evaluates to **true** if all its **Evaluations** during the grace time evaluate to **true**.
```

> If a Condition evaluates to **true**, Ops Manager lets the administrator know about that, as follows:

Reviewer: ```suggestion
If a Condition evaluates to **true**, Ops Manager lets the administrator know in the following way:
```

> * Alerts are generated, which are shown in the NOM UI as banners, alert counts and alert lists.

Reviewer: ```suggestion
* Alerts are generated and shown in the NOM UI as banners, alert counts, and alert lists.
```

> * Notifications are generated, which result in a message being sent to one of the defined targets.

Reviewer: ```suggestion
* Notifications are generated and a message is sent to one of the defined targets.
```

Conditions can be managed per DBMS in the Conditions tab of the Alerts page.

See xref:./alerts-conditions.adoc[Alerts and Conditions] on how to manage Conditions.

== Alerts

> **Alerts** are shown in the NOM UI to notify the administrator about Conditions that evaluated to **true**.

Reviewer: ```suggestion
**Alerts** are shown in the NOM UI to notify the administrator about Conditions that evaluate to **true**.
```

In case there are Alerts in the managed system, Ops Manager provides following information about them:

* Total count of Alerts is displayed in the sidebar.
* Alert count per DBMS is displayed in the DBMS dropdown (for all accessible DBMSs) and the sidebar (for the currently selected DBMS).
* Alert banner is displayed, allowing the administrator to show Alerts details or DBMSs metrics.
* Detailed alert information is available from the Alerts page of the DBMS or from the Home page (in the Alert panel of the selected DBMS).

See xref:./alerts-conditions.adoc[Alerts and Conditions] on how to manage Alerts.

== Notifications

> While Alerts are only displayed in the NOM browser page, Notifications enable sending information about triggered Conditions to external systems like email servers or other messaging systems.

Reviewer: ```suggestion
While Alerts are only displayed in the NOM browser page, Notifications enable sending information about triggered Conditions to external systems, such as email servers or other messaging systems.
```
